tests/include/mir_test_framework/window_management_test_harness.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 or 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIR_TEST_FRAMEWORK_WINDOW_MANAGEMENT_TEST_HARNESS_H
+#define MIR_TEST_FRAMEWORK_WINDOW_MANAGEMENT_TEST_HARNESS_H
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include <miral/window_management_policy.h>
+#include <miral/application.h>
+#include <miral/window_manager_tools.h>
+#include <mir/shell/surface_specification.h>
+#include "mir/test/doubles/fake_display_configuration_observer_registrar.h"
+#include "mir_test_framework/headless_in_process_server.h"
+
+namespace mir::scene
+{
+class Surface;
+class Session;
+}
+
+namespace mir_test_framework
+{
+using WindowManagementPolicyBuilder =
+    std::function<std::unique_ptr<miral::WindowManagementPolicy>(miral::WindowManagerTools const& tools)>;
+
+/// A harness for window management testing. To use, extend this class and provide the
+/// necessary virtual methods.
+class WindowManagementTestHarness : public mir_test_framework::HeadlessInProcessServer
+{
+public:
+    void SetUp() override;
+    void TearDown() override;
+
+    auto open_application(std::string const& name) -> miral::Application;
+
+    /// Create a regular application window with the provided width and height
+    auto create_window(
+        miral::Application const&,
+        mir::geometry::Width const&,
+        mir::geometry::Height const&) -> miral::Window;
```

Comment:
Do we need this alongside the more general overload? And, if we do, wouldn't `mg::Size` be clearer?

Reply:
Eh you're probably right. This was a QOL thing, but it's probably too opaque